pyomo/core/base/set.py

```diff
@@ -2200,6 +2264,26 @@ def __init__(self, *args, **kwds):
             if self._init_dimen.constant():
                 self._dimen = self._init_dimen(self.parent_block(), None)
 
+            if self._validate.__class__ is ParameterizedIndexedCallInitializer:
+                # TBD [JDS: 8/2024]: should we deprecate the "expanded
+                # tuple" version of the validate callback for scalar sets?
+                # It is widely used and we can (reasonably reliably) map to
+                # the expected behavior...
+                orig_fcn = self._validate._fcn
+                self._validate = ParameterizedScalarCallInitializer(
+                    lambda m, v: orig_fcn(m, *v), True
+                )
+
+            if self._filter.__class__ is ParameterizedIndexedCallInitializer:
+                # TBD [JDS: 8/2024]: should we deprecate the "expanded
+                # tuple" version of the filter callback for scalar sets?
+                # It is widely used and we can (reasonably reliably) map to
+                # the expected behavior...
+                orig_fcn = self._filter._fcn
+                self._filter = ParameterizedScalarCallInitializer(
+                    lambda m, v: orig_fcn(m, *v), True
+                )
+
```

Comment:
Note that this code (silently) preserves the old API for scalar Set objects.  However, that old API is
- difficult to reconcile with the "newer" API for IndexedSets
- widely used in Pyomo models in the wild
I think I'd like to put in a deprecation warning here, but that would generate a LOT of noise (basically, it would generate deprecation warnings for *every* model that uses `filter=` or `validate=`).

Thoughts (@blnicho, @mrmundt, @emma58, @michaelbynum, @whart222)?

Reply:
I don't mind disturbing users in the wild, but I want the deprecation warnings to be helpful / tell them how to change over their models. What is the new, preferred method for doing this?  Either we detail that directly in the deprecation message _or_ we make a new page on RTD that gives more detailed instructions.

Reply:
We had a long discussion at the Dev call today and decided that we should start issuing deprecation warnings for ScalarSets now.